platform/admin/core/admin-core-appservice/src/test/java/org/codice/ddf/admin/application/service/impl/ApplicationServiceBeanTest.java

```diff
@@ -174,12 +187,47 @@ public void setUpTree() throws Exception {
     @Test
     public void testInit() throws Exception {
         ApplicationServiceBean serviceBean = new ApplicationServiceBean(testAppService,
-                testConfigAdminExt);
+                testConfigAdminExt, mBeanServer);
         serviceBean.init();
 
-        //Do it again, should remove and re-init
+        verify(mBeanServer).registerMBean(serviceBean, objectName);
+    }
+
+    /**
+     * Tests the {@link ApplicationServiceBean#init()} method for the case
+     * where the serviceBean has already been initialized
+     *
+     * @throws Exception
+     */
+    @Test
+    public void testInitTwice() throws Exception {
+        ApplicationServiceBean serviceBean = new ApplicationServiceBean(testAppService,
+                testConfigAdminExt, mBeanServer);
+        when(mBeanServer.registerMBean(any(Object.class), any(ObjectName.class)))
+                .thenThrow(new InstanceAlreadyExistsException()).thenReturn(null);
+
+        serviceBean.init();
+
+        verify(mBeanServer).unregisterMBean(objectName);
```

Comment:
Sorry, forgot to mention this yesterday but you may want to add `atMost(1)` here since it's important that `unregisterMBean()` doesn't get called more than once.